### PR TITLE
macOS Tiger platform and instructions

### PR DESCRIPTION
**Pull request by glebm**
PXL_20241211_150048933

This required a few workarounds, the trickiest one in the last commit of this PR.
We can also backport this to the 1.5 branch for the 1.5.4 release.

I haven't tested netplay between big-endian and little-endian machines but I'm guessing it is still broken.

[devilutionx_tiger_ppc.dmg.zip](https://github.com/user-attachments/files/18099333/devilutionx_tiger_ppc.dmg.zip)

/cc @kencu @barracuda156 @mascguy

### CODE REVIEW

**Comment by AJenbo** [comment]
When testing with SDL 1.2.15 from Tigerbrew I noticed that it's just duping the saves and settings in the home folder so we need a Tinger/macOS addition to our implementation of SDL_GetPrefPath() in sdl2_to_1_2_backports.cpp

**Comment by glebm** [comment]
Yeah I see the issue with the paths:

image

**Comment by AJenbo** [comment]
Arh forgot about the icon, was it just the size that was the problem?

File: docs/building.md [hunk 1]
```diff
@@ -528,6 +528,44 @@ Executing `Packaging/miyoo_mini/build.sh` will create the folder `build-miyoo-mi
...
+
+http://macports-tiger-ppc.glebm.com/
+
+After installing MacPorts, run:
+
+~~~ bash
+# Some packages may require you to manually deactivate certain ports during installation.
+# Remember to reactivate them after installing.
+sudo port install curl curl-ca-bundle gcc14 cmake \
+  libsdl12 libsdl_image libsodium libfmt11 bzip2 zlib lua54 \
```
Comment (AJenbo) [inline docs/building.md:545]: libfmt11 isn't being used so no need to build it

File: docs/building.md [hunk 2]
```diff
@@ -528,6 +528,44 @@ Executing `Packaging/miyoo_mini/build.sh` will create the folder `build-miyoo-mi
...
+  google-benchmark gtest
```
Comment (AJenbo) [inline docs/building.md:546]: gtest appears to be disabled as well

Comment (AJenbo) [reply docs/building.md:546]: ... also google-benchmark isn't in my ports yet :)

Comment (glebm) [reply docs/building.md:546]: Neither google-benchmark nor gtest work from macports (google-benchmark has been merged though), I'll push a commit to use the vendored ones in a moment

File: docs/building.md (hunk 1 above)
Comment (glebm) [reply docs/building.md:545]: Yeah, also gtest and google-benchmark aren't used.
I'll removed them if the upstream issues aren't fixed by the time this is ready to merge

File: Source/utils/sdl2_to_1_2_backports.cpp [hunk 3]
```diff
@@ -897,7 +897,7 @@ char *SDL_GetPrefPath(const char *org, const char *app)
 			SDL_SetError("neither XDG_DATA_HOME nor HOME environment is set");
 			return NULL;
 		}
-#if defined(__unix__) || defined(__unix)
+#if defined(__unix__) || defined(__unix) || defined(__APPLE__)
 		append = "/.local/share/";
```
Comment (AJenbo) [inline Source/utils/sdl2_to_1_2_backports.cpp:901]: This isn't the right path for Mac, on phone ATM but check the config Wiki for an example of the Path

Comment (AJenbo) [reply Source/utils/sdl2_to_1_2_backports.cpp]: ~/Library/Application Support/diasurgical/devilution

Comment (glebm) [reply Source/utils/sdl2_to_1_2_backports.cpp]: Yep, this is now fixed in the latest version

File: docs/building.md (hunk 1 above)
Comment (glebm) [reply docs/building.md]: Done

**Comment by glebm** [comment]
@barracuda156 X11 server doesn't build from the current MacPorts due to https://trac.macports.org/ticket/62654
It's probably possible to build an older version but SDL1 appears to work well enough

With this PR, DevilutionX should be buildable from upstream MacPorts!

**Comment by barracuda156** [comment]
@glebm I see. I have probably built it only on 10.5 and 10.6 on PowerPC.
SDL2 itself builds on Tiger, but with a couple of extra disablings (and will be useless without X server anyway).

**Comment by glebm** [comment]
DMG uploaded (see PR description)!

**Comment by AJenbo** [comment]
> @glebm Have you taken a look at SDL2? As of now, it probably won’t work for games, since joystick module needs to be fixed, but video works via x11.

@barracuda156 joystick support isn't required, it does throw an error because it fails to init it, but if you ignore that it works fien. From what I read the joystick issue originated in a bug in GCC as should be solved now that we have upgraded to GCC14 so the patch that disables it can probably be removed, but we can't currently test it because of the mentioned issues around building x11.

File: docs/building.md [hunk 4]
```diff
@@ -528,6 +528,49 @@ Executing `Packaging/miyoo_mini/build.sh` will create the folder `build-miyoo-mi
...
+~~~
+
+To run tools from the `tools/` directory (only needed for development), you also need Python:
+
+~~~ bash
+sudo port install python312
+sudo port select --set python python312
+sudo port select --set python3 python312
+~~~
+
```
Comment (AJenbo) [inline docs/building.md:571]: While not an issue I have a hard time seeing who would do so :D

**Review by AJenbo — APPROVED** [review]
Oddly having a target where the threaded loading isn't working properly makes me feel the solution is more robust :+1:

**Pull request merged**